<?xml version="1.0" encoding="utf-8"?>
<launch>

  <include file="$(find third_arm)/launch/third_arm.launch"/>

  <!--include file="$(find openni_launch)/launch/openni.launch">
    <arg name="camera" value="kinect"/>
  </include -->

  <include file="$(find astra_launch)/launch/astra.launch"/>
  
  <arg name="rvizconfig" default="$(find third_arm)/urdf/apriltag_config.rviz"/>
  
  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
    <!-- Remap topic required by the node to custom topics -->
    <remap from="image_rect" to="camera/rgb/image_rect_color" />
    <remap from="camera_info" to="camera/rgb/camera_info" />

    <!-- Optional: Subscribe to the compressed stream-->
    <param name="image_transport" type="str" value="compressed" />

    <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
    <param name="tag_family" type="str" value="36h11" />

    <!-- Enable projected optical measurements for more accurate tag transformations -->
    <!-- This exists for backwards compatability and should be left true for new setups -->
    <param name="projected_optics" type="bool" value="true" />

    <!-- Describe the tags -->
    <rosparam param="tag_descriptions">[      
      {id: 1, size: 0.09, frame_id: base_frame},
      {id: 2, size: 0.09, frame_id: ee_frame},
      {id: 12, size: 0.07, frame_id: cup_frame},      
      {id: 3, size: 0.09, frame_id: left_hand},
      {id: 4, size: 0.09, frame_id: right_hand}]      
    </rosparam>
  </node>

  <!--include file="$(find skeleton_markers)/launch/markers_from_tf.launch"/-->
  <!-- node name = "hrc_2d_poser" pkg="third_arm" type="hrc2d_apriltags_2d_ik.py" output ='screen'/-->
  

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="false"/>

  <node name="hrc2d_gui" pkg="third_arm" type="hrc2d_gui_main.py" output='screen' />

</launch>